File: SWEA/D3/1220. ［S／W 문제해결 기본］ 5일차 － Magnetic/［S／W 문제해결 기본］ 5일차 － Magnetic.py


for tc in range(1, 11):
    n = int(input())

    arr = [list(map(int, input().split())) for _ in range(100)]


    #위 n극 아래 s 극
    answer = 0
    for c in range(100):
        red = False

        #down
        for r in range(100):
            if arr[r][c] == 1:
                red = True
            elif arr[r][c] == 2:
                if red:
                    answer += 1
                    red = False

    print(f"#{tc} {answer}")

File: SWEA/D3/1220. ［S／W 문제해결 기본］ 5일차 － Magnetic/［S／W 문제해결 기본］ 5일차 － Magnetic.java
import java.util.Scanner;


class Solution {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		StringBuilder sb = new StringBuilder();
		
		for(int tc=1; tc<=10; tc++) {
			int N = sc.nextInt();
			
			int[][] arr = new int[100][100];
			for(int r=0; r<100; r++) {
				for(int c=0; c<100; c++) {
					arr[r][c] = sc.nextInt();
				}
			}
			int answer = 0;
			boolean red = false;
			for(int c=0; c<100; c++) {
				red = false;
				
				for(int r=0; r<100; r++) {
					if(arr[r][c] == 1) red = true;
					else if(arr[r][c]==2) {
						if(red) {
							answer++;
							red = false;
						}
					}
				}
			}
			sb.append("#").append(tc).append(" ").append(answer).append("\n");
		}
		System.out.println(sb.toString());
	}
	
		
	
}
